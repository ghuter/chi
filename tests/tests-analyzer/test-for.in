f :: fun () {
	x :: 10;
	for i := 0; i < x; i += 1 {
		x :: 1;
	}
}
