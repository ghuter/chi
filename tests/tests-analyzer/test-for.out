SFUN(f, {SSEQ(const SDECL(x, ECSTI(10)) : i32, SFOR(SDECL(i, ECSTI(0)) : i32, EBINOP(LT, EMEM(i) : i32, EMEM(x) : i32) : bool, SASSIGN(i, EBINOP(ADD, EMEM(i) : i32, ECSTI(1)) : i32), const SDECL(x, ECSTI(1)) : i32))}) : void
